utils/port.go

```diff
@@ -18,3 +24,61 @@ func GetFreePort() (int, error) {
 	defer l.Close()
 	return l.Addr().(*net.TCPAddr).Port, nil
 }
+
+// FindPortInRange Iterate through CSV or Range of ports to find open port
+// Valid inputs are "8081", "8081,8085", "8081-8085". Do not combine
+// list and range
+func FindPortInRange(s string) (int, error) {
+	// Take care of csv and single value
+	if !strings.Contains(s, "-") {
+		ports := strings.Split(strings.TrimSpace(s), ",")
+		for _, p := range ports {
+			i, err := strconv.Atoi(p)
+			if err != nil {
+				return 0, err
+			}
+			err = checkPort(i)
+			if err != nil {
+				continue
+			}
+			return i, nil
+		}
+		return 0, errors.New("all passed ports are used")
+	}
+	// Now take care of ranges
+	ports := strings.Split(strings.TrimSpace(s), "-")
+	if len(ports) != 2 {
+		return 0, errors.New("invalid range passed")
+	}
+	lower, err := strconv.Atoi(ports[0])
+	if err != nil {
+		return 0, err
+	}
+	upper, err := strconv.Atoi(ports[1])
+	if err != nil {
+		return 0, err
+	}
+	for i := lower; i <= upper; i++ {
```

Comment:
Are we missing a test case here? If somebody passes in `8888-7777` should we return something like `invalid range passed` as per above?